runtime/src/serde_snapshot.rs

```diff
@@ -372,17 +376,19 @@ fn reconstruct_single_storage<E>(
     slot: &Slot,
     append_vec_path: &Path,
     storage_entry: &E,
+    new_append_vec_id: Option<AppendVecId>,
```

Comment:
If `SerializableStorage` supported a `set_append_vec_id()` function, maybe we could just set the `new_append_vec_id` before this function is called and remove this additional argument?

Reply:
This is awesome! I really wanted this to work, but unfortunately I don't think it will be worth it...

Since the `snapshot_storages` (and therefore `slot_storage`) is inside a parallel loop, if I want to mutate `SerializableStorage` that would require `E` (and therefore `SerializableStorage`) to be `Send` everywhere. I went down this rabbit hole for a while, adding `Send` to everything, until I got to the serde tests and the errors started to get more involved.

Check out the last line for what I needed to do, in this error here:
```
error[E0277]: `<C as serde_snapshot::TypeContext<'_>>::SerializableAccountStorageEntry` cannot be sent between threads safely
   --> runtime/src/serde_snapshot/tests.rs:76:5
    |
76  |     reconstruct_accountsdb_from_fields(
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `<C as serde_snapshot::TypeContext<'_>>::SerializableAccountStorageEntry` cannot be sent between threads safely
    |
    = help: the trait `Send` is not implemented for `<C as serde_snapshot::TypeContext<'_>>::SerializableAccountStorageEntry`
note: required by a bound in `serde_snapshot::reconstruct_accountsdb_from_fields`
   --> runtime/src/serde_snapshot.rs:407:50
    |
394 | fn reconstruct_accountsdb_from_fields<E>(
    |    ---------------------------------- required by a bound in this
...
407 |     E: SerializableStorage + std::marker::Sync + std::marker::Send,
    |                                                  ^^^^^^^^^^^^^^^^^ required by this bound in `serde_snapshot::reconstruct_accountsdb_from_fields`
help: consider further restricting the associated type
    |
68  |     R: Read, <C as serde_snapshot::TypeContext<'_>>::SerializableAccountStorageEntry: Send
    |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I could also solve this by:
1. Putting the mutation in an unsafe block, or
2. Cloning the `slot_storage` each iteration

but both those options seem bad. It feels like the current impl is the easiest. What do you think?